### DISCUSSION TITLE
Improved Examples, timeout management and LastError messages

### DISCUSSION THREAD

**Discussion by gtbyio** [Makuna/Rtc#153] [General]
following the last topic, a request to take into account
 of the wire changes for the setWireTimeout() given in the Examples.
and also the error messages given by Rtc.LastError

thanks Makuna

**Comment by gtbyio** [comment]
... a possible improvement, 

Interuption management:
the for test (Arduino/Uno) only 2 Int :  Pin 2 or 3 of the board  
 
in my example, a TTL/RS485 Module has the possibility to use manually one of the two Pin

 #define MY_RS485_DE_PIN 3 (or 2) //cmd DE pin
 
 but for the RtcDS3231 library, as far as I understood, it is automatic ?
 
 Well, with my power failure test, the result gives me a value of Rtc.LastError always at Zero.
 But if I disconnect the Pin x of the RS485 Module, the value goes to 5.
 
 I thought that it was necessary to force the value of the Int with a cabling, the SQW of the DS3231 and put on Pin2 (the reverse of the RS485).
 But no, still no change on Rtc.LastError
 
 This is not very important, but it should be said :) 
 
 Thanks Makuna for your work

**Reply by Makuna** [reply]
This library will use the WIRE library to handle the low level hardware interface (I2C orTWI), and it only does two pins, SDA, SCL.  So, nothing is managing the DE or RE as those would be RS485 specific.  On some boards like AVR, WIRE interfaces with hardware which has pin restrictions.  I believe it does something similiar by holding one of the pins low so other devices notice they should listen.  I2C is a controller and peripheral interface, unlike RS485 which allows anyone attached to start communications to any other device.


I don't know what the right way to wire the RS485 module when using it without DE/RE features.  You could always wrap WIRE with a class that implements the extra two as my library allows any class and instance that uses the same API as WIRE.

**Comment by gtbyio** [comment]
Thanks for the explanation.
problem solved 
https://www.mysensors.org/build/rs485
